xjc-maven-plugin/src/main/java/com/github/davidmoten/xjc/XjcMojo.java

```diff
@@ -145,6 +163,37 @@ public void execute() throws MojoExecutionException, MojoFailureException {
         return command;
     }
 
+    private Stream<String> getPluginRuntimeDependencyEntries()
+    {
+        PluginDescriptor pluginDescriptor = (PluginDescriptor) getPluginContext().get(PLUGIN_DESCRIPTOR);
+        Plugin plugin = mavenProject.getBuild().getPluginsAsMap().get(pluginDescriptor.getPluginLookupKey());
+
+        List<ArtifactResolutionResult> artifactResolutionResults = plugin.getDependencies().stream()
+            .map(repositorySystem::createDependencyArtifact)
+            .map(this::resolve)
+            .collect(Collectors.toList());
+
+        Stream<Artifact> originalArtifacts = artifactResolutionResults.stream()
+            .map(ArtifactResolutionResult::getOriginatingArtifact);
+
+        Stream<Artifact> childArtifacts = artifactResolutionResults.stream()
+            .flatMap(resolutionResult -> resolutionResult.getArtifactResolutionNodes().stream())
+            .map(ResolutionNode::getArtifact);
+
+        return Stream.concat(originalArtifacts, childArtifacts)
+            .map(Artifact::getFile)
+            .map(File::getAbsolutePath);
+    }
+
+    private void addGeneratedSourcesToProjectSourceRoot() {
```

Comment:
Good idea to do this. Unfortunately I'll probably still use a redundant build-helper-maven-plugin in my projects so that IDEs know that there is generated source around.

Reply:
Are we assuming too much always adding to the source root? I guess we can make it configurable later if someone complains.

Reply:
Ok, will revert this behavior